src/MetaMorpho.sol

```diff
@@ -367,12 +369,19 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
                 if (config[id].cap != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroCap(id);
 
-                if (MORPHO.supplyShares(id, address(this)) != 0) {
+                uint256 supplyShares = MORPHO.supplyShares(id, address(this));
+
+                if (supplyShares != 0) {
                     if (config[id].removableAt == 0) revert ErrorsLib.InvalidMarketRemovalNonZeroSupply(id);
 
                     if (block.timestamp < config[id].removableAt) {
                         revert ErrorsLib.InvalidMarketRemovalTimelockNotElapsed(id);
                     }
+
+                    uint256 totalSupplyAssets = MORPHO.totalSupplyAssets(id);
```

Comment:
```suggestion
                   // Not accruing interest, to avoid subtracting too much interest from `lastTotalAssets` than necessary.
                    uint256 totalSupplyAssets = MORPHO.totalSupplyAssets(id);
```
Since it's not obvious why we're not using the expected supply assets balance.

Reply:
It would revert if we'd call `expectedSupplyAssets` right ?

Reply:
```suggestion
                    // Not accruing interest to avoid reverting.
                    uint256 totalSupplyAssets = MORPHO.totalSupplyAssets(id);
```

Reply:
```suggestion
                    // Not accruing interest to avoid possible reverts due to a broken IRM.
                    uint256 totalSupplyAssets = MORPHO.totalSupplyAssets(id);
```

Reply:
Sorry for the back and forth, I think i prefer being specific with my latest suggestion

Reply:
Indeed!